src/lib/markbind/src/lib/markdown-it/markdown-it-date.js

```diff
@@ -0,0 +1,97 @@
+const Moment = require('moment');
+
+const BASE_FORMAT = 'YYYY-MM-DD';
+const EXPECTED_FORMAT = `@date:${BASE_FORMAT} week#day@`;
+const STRICT_MODE = true;
+
+const BASE_INVALID = `Base date format must be ${BASE_FORMAT}, instead received`;
+const ERR_MSG = `Unexpected format, expected ${EXPECTED_FORMAT}, instead received`;
+const DAY_INVALID = 'Unexpected day identifier';
+const DAY_OUT_OF_RANGE = 'Day number should be between 1 - 7 inclusive';
+
+const DEFAULT_FORMAT = 'ddd D MMM'; // e.g. Thu 27 Aug
+const REGEX_LOOSE = /@date[\w\W]*?@/;
+const REGEX_STRICT = /@date(?::)(\d{4}-\d{2}-\d{2})\s[wW]?(\d+)#(\d|\w{3})(?:\s?\|\s?([^@]*?))?@/;
+
+const days = 'mon tue wed thu fri sat sun'.split(' ');
+
+module.exports = require('markdown-it-regexp')(
+  REGEX_LOOSE,
+  (match) => {
+    let raw = match[0];
+
+    const matched = raw.match(REGEX_STRICT);
+
+    // Invalid format
+    if (!matched) {
+      return formatError(ERR_MSG, raw).err;
+    }
+
+    const {base, day, week, formatPattern, err} = parseInput(raw, matched);
+    if (err) return err;
+
+    return calculateTarget(base, week, day).format(formatPattern)
+  }
+);
+
+/**
+ * Parses and extracts the week, day offset and format from the raw matched expression
+ * @param raw raw matched expression
+ * @param match matched token groups
+ */
+function parseInput(raw, match) {
+  // Parse base date
+  let base;
+  try {
+    base = Moment(match[1], BASE_FORMAT, STRICT_MODE);
+  } catch (e) {
+    return formatError(BASE_INVALID, match[1]);
+  }
+
+  // Parse week number
+  const week = parseInt(match[2]);
+
+  // Parse day
+  let day = match[3].toLowerCase();
+  if (isNaN(day)) {
+    if (days.includes(day)) {
+      // Maps 'mon' => 1, 'tue' => 2 and so on
+      day = days.indexOf(day) + 1;
+    } else {
+      return formatError(DAY_INVALID, raw);
+    }
+  } else {
+    day = parseInt(day);
+    if (day < 1 || day > 7) {
+      return formatError(DAY_OUT_OF_RANGE, raw);
+    }
+  }
+
+  let formatPattern = match[4] || DEFAULT_FORMAT;
+  formatPattern = formatPattern === '' ? DEFAULT_FORMAT : formatPattern;
```

Comment:
Should be able to reduce to
```suggestion
  const formatPattern = match[4] === '' ? DEFAULT_FORMAT : formatPattern;
```
?

Reply:
match[4] could be empty string or undefined